core/node/infra/metrics_factory.go

```diff
@@ -39,6 +41,10 @@ type MetricsFactory interface {
 	NewStatusCounterVecEx(name string, help string, labels ...string) *StatusCounterVec
 
 	Registry() *prometheus.Registry
+
+	// GetMetricsAsMap returns all metrics as a structured map suitable for JSON logging.
+	// This is intended for debugging purposes only.
+	GetMetricsAsMap() (map[string]interface{}, error)
```

Comment:
> This is a work in progress; however I'd still like to capture it in the repository for later use.

Do you want to merge this at some point? This function is never used.

What I sometimes do with functions that intended for debugging/testing is to put them in a separate interface. So in this case:
```
type (
   MetricsFactory interface { ... }
   MetricsFactoryDebug interface { ... }
)
```

That prevents putting functions that are only meant for testing/debugging in production interfaces and makes it clear that these functions are not for use in production flow.